src/App.tsx

```diff
@@ -14,11 +15,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+enum Operation {
+  Add,
+  Remove,
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  goodsOperationsHandler = (operation: Operation, good: string) => {
+    switch (operation) {
+      case Operation.Add:
+        this.setState((state) => ({ selectedGoods: [...state.selectedGoods, good] }));
+
+        break;
+      case Operation.Remove:
+        this.setState((state) => (
+          { selectedGoods: state.selectedGoods.filter(goodItem => goodItem !== good) }
+        ));
+
+        break;
+      default:
+    }
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+    const goodsLengthCheck = selectedGoods.length > 1
+      ? `${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods[selectedGoods.length - 1]} are selected`
+      : `${selectedGoods[0]} is selected`;
+
+    return (
+      <div className="App">
+        <div className="App__top">
+          <h1>
+            {selectedGoods.length ? goodsLengthCheck : 'No goods selected'}
+          </h1>
+          {!!selectedGoods.length && (
+            <button
+              type="button"
+              onClick={() => {
+                this.setState({ selectedGoods: [] });
+              }}
+              onKeyDown={() => {
+                this.setState({ selectedGoods: [] });
+              }}
+            >
+              X
+            </button>
+          )}
+        </div>
+        <ul>
+          {goodsFromServer.map((good) => (
+            <li key={good}>
+              <div className="App__good">
+                <p className={classNames('App__good-text', { 'App__good-text-selected': selectedGoods.includes(good) })}>
+                  {good}
+                </p>
+                {selectedGoods.includes(good)
+                  ? (
+                    <button
+                      type="button"
+                      onClick={() => {
+                        this.goodsOperationsHandler(Operation.Remove, good);
+                      }}
+                      onKeyDown={() => {
+                        this.goodsOperationsHandler(Operation.Remove, good);
+                      }}
+                    >
+                      Remove
+                    </button>
+                  )
+                  : (
+                    <button
```

Comment:
You can create only one button and conditionally change the function and the text

Reply:
I've actually thought about doing that but two or more searches in array wouldn't be as quick as only one, anyways there's not that much items in array and after your suggestion i really think thank we can sacrifice a tiny bit of perfomance to make our code more readable. So, thank you 🔥

Reply:
Omg, i'm a bit dumb, nevermind 🤣 I've actually understood that we can make variable with a result of `includes` and it will be as productive as before))